### PR TITLE
Add docs for optimized commits for object stores

### PR DESCRIPTION
**Pull request by calvinjia**
Documents the newly added async through + persist on rename feature.

### CODE REVIEW

File: docs/en/advanced/Performance-Tuning.md [hunk 1]
```diff
@@ -234,3 +234,41 @@ When enabled, the same data blocks are available across multiple workers,
...
 dataset is large compared to the capacity of a single Alluxio worker.
+
+### Optimized Commits for Compute Frameworks
+
+Running with optimized commits through Alluxio can provide an order of magnitude improvement in the
+overall runtime of compute jobs.
+
+Computation frameworks that leverage the Hadoop MapReduce committer pattern (ie. Spark, Hive) are
+not optimally designed for interacting with storages that provide slow renames (mainly Object
+Stores). This is most common when using stacks such as Spark on S3 or Hive on Ceph.
```
Comment (LuQQiu) [inline docs/en/advanced/Performance-Tuning.md]: Could we add an example for one of the stack? The example can be a step by step paragraph or a flow chart.
And a comparison between our previous way (`CACHE_THROUGH`) and optimized way (`ASYNC_THROUGH`).
The comparison can help users have a more direct informative understanding of why we introduced those parameters.

Comment (calvinjia) [reply docs/en/advanced/Performance-Tuning.md]: Updated, thanks.

File: docs/en/advanced/Performance-Tuning.md [hunk 2]
```diff
@@ -234,3 +234,41 @@ When enabled, the same data blocks are available across multiple workers,
...
+
+The Hadoop MapReduce committer leverages renames to commit data from a staging directory (usually
+`output/_temporary`) to the final output directory (ie. `output`). When writing data with
+`CACHE_THROUGH` or `THROUGH` this protocol translates to writing the data once to Alluxio (fast) and
+object store (slow) and an additional copy and delete (rename) in the object store (slow). When
```
[the remark this answers is not in the document]
Comment (calvinjia) [reply docs/en/advanced/Performance-Tuning.md]: Updated, thanks.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**